QuizApp/TabBarController.swift

```diff
@@ -0,0 +1,24 @@
+import Foundation
+import UIKit
+
+class TabBarController: UITabBarController {
+//    var viewControllers: [UIViewController]!
+
+    convenience init(_ viewControllers: [UIViewController]) {
+        self.init()
+        self.viewControllers = viewControllers
+        setupTabBar()
+    }
+
+    func setupTabBar() {
+
+        self.viewControllers![0].tabBarItem = UITabBarItem(title: "Quiz", image: UIImage(systemName: "stopwatch"), selectedImage: UIImage(systemName: "stopwatch.fill"))
+        self.viewControllers![1].tabBarItem = UITabBarItem(title: "Search", image: UIImage(systemName: "magnifyingglass"), selectedImage: UIImage(systemName: "magnifyingglass"))
+        self.viewControllers![2].tabBarItem = UITabBarItem(title: "Settings", image: UIImage(systemName: "gearshape"), selectedImage: UIImage(systemName: "gearshape.fill"))
```

Comment:
This is not the way to set tab bar items. Each view controller can set it for itself, an you can do that in the router function.
Here you're force unwrapping an array and trying to access element at hardcoded index which can lead to crashes.